### PR TITLE
Log in front

### PR DESCRIPTION
**Pull request by Igork777**

### CODE REVIEW

File: app/src/main/java/com/example/faunaapp/view/fragments/LogInFragment.java [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+import com.google.android.material.textfield.TextInputLayout;
+
+import java.util.Objects;
+
+public class LogInFragment extends Fragment {
+
+    private Button logInButton;
+    private TextInputLayout email, password;
+    private LogInFragmentViewModel logInViewModel;
+    private View logInView;
```
Comment (jorgenfaunaapp) [inline app/src/main/java/com/example/faunaapp/view/fragments/LogInFragment.java:32]: Are you actively using the private View? If not do not declare it.

File: app/src/main/res/layout/log_in_fragment.xml [hunk 2]
```diff
@@ -32,7 +31,8 @@
         app:layout_constraintTop_toTopOf="parent" />
 
     <com.google.android.material.textfield.TextInputLayout
-        android:id="@+id/textInputLayout2"
+        android:id="@+id/emailInput"
+        android:inputType="textNoSuggestions"
         android:layout_width="220dp"
         android:layout_height="52dp"
         android:hint="@string/email"
```
Comment (jorgenfaunaapp) [inline app/src/main/res/layout/log_in_fragment.xml:38]: Consider naming your view elements after the view they are in, and use that in your translation strings eg: fragment_login_text_input_email. the translation string would then be @string/fragment_login_text_input_email

**Review by jorgenfaunaapp — COMMENTED** [review]
Would be nicer if you did com.fauna as your package.
Be carefull with naming, it should be login here, not log in.

File: app/src/main/java/com/example/faunaapp/repository/Repository.java [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+import org.jetbrains.annotations.NotNull;
+
+
+public class Repository {
+    private static Repository instance;
+    private LogInAsyncTask logInAsyncTask;
+    private static ApolloClient apolloClient;
+
+    public Repository() {
+        apolloClient = ApolloClient.builder().serverUrl("https://fauna-android-feature-t-cillum.herokuapp.com/").build();
```
Comment (jorgenfaunaapp) [inline app/src/main/java/com/example/faunaapp/repository/Repository.java:21]: Consider adding a helper to instantiate your dependencies, so you can reuse your ApolloClient instance across repositories.

**Comment by jorgenfaunaapp** [comment]
Consider separating Apollo instantiation. Consider renaming log_in to login, and look at naming of your individual view components.

File: app/src/main/java/com/example/faunaapp/view/fragments/LogInFragment.java (hunk 1 above)
Comment (Igork777) [reply app/src/main/java/com/example/faunaapp/view/fragments/LogInFragment.java:32]: I do use logInView in methods : onCreateView, initializeFragmentsValue, and in setSnackbar

**Pull request merged**